File: CodeGenDotNet 159059-1372 (a)/CSharp/Chapter 1/HelloWorldAndSimpleClass/HelloWorldViaCodeDOM.cs
using System;

/// <summary>
/// Hello World via the CodeDOM
/// </summary>
public class HelloWorldViaCodeDOM
{
	#region Class level declarations - empty
	#endregion

	#region Constructors - empty
	#endregion

	#region Public Methods and Properties
	public static System.CodeDom.CodeCompileUnit BuildGraph()
	{
		System.CodeDom.CodeCompileUnit CompileUnit = new System.CodeDom.CodeCompileUnit();

		System.CodeDom.CodeNamespace nSpace = new System.CodeDom.CodeNamespace( "HelloWorldViaCodeDOM" );
		CompileUnit.Namespaces.Add( nSpace );

		nSpace.Imports.Add( new System.CodeDom.CodeNamespaceImport( "System" ) );
		
		System.CodeDom.CodeTypeDeclaration clsStartup = new System.CodeDom.CodeTypeDeclaration( "Startup" );
		nSpace.Types.Add( clsStartup );

		System.CodeDom.CodeEntryPointMethod main = new System.CodeDom.CodeEntryPointMethod();
		System.CodeDom.CodePrimitiveExpression exp = new System.CodeDom.CodePrimitiveExpression( "Hello World!" );
		System.CodeDom.CodeTypeReferenceExpression refExp = new System.CodeDom.CodeTypeReferenceExpression( "System.Console" );
		System.CodeDom.CodeMethodInvokeExpression invoke = new System.CodeDom.CodeMethodInvokeExpression( refExp, "WriteLine", exp );
		main.Statements.Add( new System.CodeDom.CodeExpressionStatement( invoke ) );

		clsStartup.Members.Add( main );

		return CompileUnit;
	}
	#endregion

	#region Protected and Friend Methods and Properties - empty
	#endregion

	#region Private Methods and Properties - empty
	#endregion
}

File: CodeGenDotNet 159059-1372 (a)/VBCode/Chapter 1/HelloWorldAndSimpleClass/HelloWorldViaCodeDOM.vb
Option Strict On
Option Explicit On

Imports System

'! Class Summary: 

Public Class HelloWorldViaCodeDOM

#Region "Class level declarations - empty"
#End Region

#Region "Constructors - empty"
#End Region

#Region "Public Methods and Properties"
   Public Shared Function BuildGraph() As CodeDom.CodeCompileUnit
      Dim CompileUnit As New CodeDom.CodeCompileUnit

      Dim nSpace As New CodeDom.CodeNamespace("HelloWorldViaCodeDOM")
      CompileUnit.Namespaces.Add(nSpace)

      nSpace.Imports.Add(New CodeDom.CodeNamespaceImport("System"))

      Dim clsStartup As New CodeDom.CodeTypeDeclaration("Startup")
      nSpace.Types.Add(clsStartup)

      Dim main As New CodeDom.CodeEntryPointMethod
      Dim exp As New CodeDom.CodePrimitiveExpression("Hello World!")
      Dim refExp As New CodeDom.CodeTypeReferenceExpression("System.Console")
      Dim invoke As New CodeDom.CodeMethodInvokeExpression(refExp, "WriteLine", exp)
      main.Statements.Add(New CodeDom.CodeExpressionStatement(invoke))

      clsStartup.Members.Add(main)

      Return CompileUnit
   End Function
#End Region

#Region "Protected and Friend Methods and Properties - empty"
#End Region

#Region "Private Methods and Properties - empty"
#End Region

End Class
